### ISSUE TITLE
ColorVariable should support colors in maps and rgba function calls

### ISSUE THREAD

**Issue by apfelbox** [brigade/scss-lint#402]
This is more a request for comments, as both these cases are arguably. Both of them are currently reported as errors, although I would argue, that they shouldn't be. If there is no concesus, these could be added as options to the linter.

### Static color definition with alpha channel
```scss
$color-with-alpha: rgba(#000, .5);
```

### Maps containing colors
```scss
$shades-of-gray: (#4c4c4c, #626262, #7d7d7d, #979797);
```

**Comment by yeeyang** [comment]
I was just looking at this as well. Use case is when I'm looping through my map.

```
// example
$themes-map: (
   some-theme: (
      'red': #f00,
      'green': #458b00
   ),
   ...
);
```

The key of `'red'` and `'green'` will return as a false positive as trying to declare a color literal. I'm probably doing something wrong anyway but wanted to report my findings.

**Comment by sds** [comment]
Please see the discussion in https://github.com/causes/scss-lint/issues/175#issuecomment-45644321 for why using `red` and `green` for key names is a bad idea.

**Comment by apfelbox** [comment]
@sds the comment about color names as map keys is totally unrelated to my issue. Please see both my code examples.

**Comment by apfelbox** [comment]
@yeeyang this is not related to this issue, please see the comment as linked by @sds.

**Comment by sds** [comment]
Sorry, @apfelbox. I've seen a number of similar issues reported so I accidentally thought yours similar.

I'm not exactly sure what answer to offer here, as it's hard to come up with a solution that is exhaustive. I suppose we could add some option to allow specifying literals in maps, but then why don't you just use control comments to disable linting on the map declaration? For example:

```scss
// scss-lint:disable ColorVariable
$shades-of-gray: (#4c4c4c, #626262, #7d7d7d, #979797);
// scss-lint:enable ColorVariable
```

This is what control comments were meant for, so it's not clear to me why we need to start treating these special cases directly. Happy to hear your thoughts.

**Comment by apfelbox** [comment]
No worries :wink: 

I always see comments to temporarily disable the linting as a hack until some very complex cases are properly supported by the regular linting rules or the other way around: as a hack in the code, that I don't want to follow the linting rules in this particular piece of code instead of properly refactoring it.

But I think that maps and lists are common enough data structures, that they should be handled properly. Especially as the linter `ColorVariable` (in my understanding) implies that the color needs to be declared in a *variable*, it does not specify in *which type* of variable (scalar, list or map).

About `rgba()`: this is both a built-in function by Sass, and a CSS defined "pseudofunction". Sass just adds the convenience to not just support `rgba(255, 255, 255, .5)` but also `rgba(#fff, .5)`. Although this is technically a function call and therefore no declaration in a variable (but instead in a function argument), I think it would be the better way to just explicitly this function call.

```scss
$v1: rgba(255, 255, 255, .5); // accepted
$v2: rgba(#fff, .5); // marked as warning
```

Although scss-lint is technically correct here, I would argue that the second version is not wrong and should also be enabled. What do you think?

**Comment by sds** [comment]
Those seem like good reasons to me. I think it makes sense for `ColorVariable` to support the two cases you describe, namely:

- Color literals in map declarations should not report a lint
- Color literals in the `rgba(#fff, .5)` helper should not report a lint

I'll get to these when I can, but otherwise a pull request with tests would be welcome!

**Comment by sds** [comment]
Implemented in 4b91c9e.

**Comment by apfelbox** [comment]
@sds nice work, thanks!
